src/service/viewer-impl.js

```diff
@@ -276,6 +276,16 @@ export class Viewer {
     this.performanceTracking_ = this.params_['csi'] === '1';
     dev.fine(TAG_, '- performanceTracking:', this.performanceTracking_);
 
+    /**
+     * Whether the AMP document is embedded in a viewer, such as an iframe or
+     * a web view.
+     * Notice that the value of the "origin" parameter itself is never consulted
+     * here, but instead confirmed via handshake in the `setMessageDeliverer`
+     * method.
+     * @private @const {boolean}
+     */
+    this.isEmbedded_ = this.isIframed_ || !!this.params_['origin'];
```

Comment:
Currently, the consequence of being embedded is a trusted mode. Not much constraints we can reliably apply given that web view allows arbitrary script injection. We're currently going through security review with this. I can add an experiment if you'd like. But, again, the protection from it will be minimal.